docs/gossip.md

```diff
@@ -23,4 +24,82 @@ If you are using [bastion hosts](bastion.md), it is a bit tricky to find the bas
 
 ```
 kops toolbox dump -ojson | grep 'bastion.*elb.amazonaws.com'
-```
\ No newline at end of file
+```
+
+## Switching to memberlist from weaveworks mesh
+
+Weaveworks mesh is known to spike exponentially with CPU, memory and network bandwidth usage on cluster scale up. [issue 7429](https://github.com/kubernetes/kops/issues/7427), [issue 7436](https://github.com/kubernetes/kops/issues/7436), [issue 13974](https://github.com/kubernetes/kops/issues/13974)
+
+### Turn on memberlist as secondary protocol
+
+Add `gossipConfig` to your cluster configuration:
+
+```bash
+kops edit cluster
+```
+
+```yaml
+  gossipConfig:
+    listen: 0.0.0.0:3999
+    protocol: mesh
+    secondary:
+      listen: 0.0.0.0:4000
+      protocol: memberlist
+
+  dnsControllerGossipConfig:
+    protocol: mesh
+    listen: 0.0.0.0:3998
+    seed: 127.0.0.1:3999
+    secondary:
+      protocol: memberlist
+      listen: 0.0.0.0:3993
+```
+
+Perform a rolling-update of control-plane nodes only.
+
+Check that dns-controller is restarted and now uses the new gossip protocol (you can see that under args).
+
+```bash
+kubectl get pod -n kube-system | grep dns-controller
+kubectl get pod -n kube-system | grep dns-controller | awk '{print $1}' | xargs kubectl get pod -nkube-system -oyaml
+```
+
+```
+  - args:
+    - --watch-ingress=false
+    - --dns=gossip
+    - --gossip-protocol=mesh
+    - --gossip-listen=0.0.0.0:3998
+    - --gossip-seed=127.0.0.1:3999
+    - --gossip-protocol-secondary=memberlist
+    - --gossip-seed-secondary=127.0.0.1:3993
+    - --zone=staging.stats.superbet.com
+    - --internal-ipv4
+    - --zone=*/*
+    - -v=2
+```
+
+### 2 Switch memberlist to primary and remove secondary protocol
+
+```yaml
+  gossipConfig:
+    protocol: memberlist
+    listen: 0.0.0.0:4000
+    secondary:
+      protocol: ""
+
+  dnsControllerGossipConfig:
+    protocol: memberlist
+    listen: 0.0.0.0:3993
+    seed: 127.0.0.1:4000
+    secondary:
+      protocol: ""
+```
+
+Restart worker nodes and then restart control-plane nodes
+
+Note that this will not flag nodes as *Needs update*; you will have to do *rolling-update* with `--force` flag.
+
+**!Restart control-plane nodes last, othwerwise all nodes that still have mesh as primary will get into NotReady state!**
+
+**Do not remove secondary protocol completly, otherwise new nodes won't join** [duplicate metrics issue, Protokube won't ever start](https://github.com/kubernetes/kops/issues/9006)
```

Comment:
What's going on here? The secondary protocol is defaulting to something? Can kOps not be fixed to avoid this problem?

Reply:
As I understand, the problem is with the Prometheus metrics for some reason. I haven't investigate to be honest.

As already suggested here in comments, the idea is of removing both of these protocols so I am not sure if this is worth fixing. 

Main reason I think this i worth documenting is to note that there could be issues with current gossip protocol and to suggest temporary fix for it. 
It is difficult to find this problem when some nodes randomly start OOMing and you have to spend quite some time debugging to find that the protokube gossip protocol is the the problem. There are Github issues about that, but it's hard to find them when you don't know what you are searching for.

Therefore I think this is worth documenting, but not worth spending time to simplify this through some kOps optimizations.